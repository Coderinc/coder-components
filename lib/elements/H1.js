import styled from 'styled-components';

export default styled.h1`
  color: ${props => props.theme.primaryFont};
  font-size: 2.25rem;
  line-height: 3.375rem;
  font-weight: ${props => (props.light ? 'lighter' : 'bold')};
`;

