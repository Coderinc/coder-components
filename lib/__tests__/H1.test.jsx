import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import H1 from '../elements/H1';

describe('H1', () => {
  it('renders correctly', () => {
    const component = renderer.create(<H1>Lorem Ipsum</H1>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('light font', () => {
    const component = renderer.create(<H1 light>Lorem Ipsum</H1>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});