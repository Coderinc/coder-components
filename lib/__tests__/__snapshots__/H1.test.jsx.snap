// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 light font 1`] = `
.c0 {
  font-size: 2.25rem;
  line-height: 3.375rem;
  font-weight: lighter;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`H1 renders correctly 1`] = `
.c0 {
  font-size: 2.25rem;
  line-height: 3.375rem;
  font-weight: bold;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;
